server-timing/defaults.php

```diff
@@ -152,6 +152,23 @@ function( $passthrough = null ) {
 		PHP_INT_MAX
 	);
 
+	add_action(
```

Comment:
I'd prefer to see this implemented as a named callback that could more easily be removed on sites that want to run the plugin but not expose this data. Also, I think we should be consistent about prefixing our metrics with `wp-` or similar, to avoid potential collisions if other tools are adding data to the Server Timing header.

I also notice that the [MDN docs for Server Timing](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing) suggest using a semi-colon separated syntax for reporting these metrics. Not sure if this is a convention that we should be following or not, but it would suggest that we do something like:

```
Server-Timing: wp-total;desc="WordPress Total";dur={value}
```
or (simplified)

```
Server-Timing: wp-total;dur={value}
```

Curious what your opinion is of those conventions.

Reply:
@joemcgill Replying to your points:
* If there is a need to remove metrics, I'd prefer we handle that holistically in a separate issue. The Server-Timing header is controlled via an API, and I'm not sure that removing hooks is the cleanest way to go about it. More importantly though, for this specific metric, I don't see why someone would want to remove `wp-total` when that's the same as `wp-before-template + wp-template` anyway, which would still not be possible to be removed. That's why I think we should rather approach it holistically.
* The Server-Timing API enforces the `wp-` prefix, so it will show as you mention :)
* Similarly, the Server-Timing API controls the exact output, and it will be output exactly like you're saying, following the conventions (your 2nd example, since we don't use metric descriptions at this point).

Reply:
Thanks for your responses, @felixarntz. I had wondered if points 2 and 3 were handled elsewhere in the process. Glad to know that it is. I do think we should revisit how certain metrics can be managed and am fine for that to be done outside the scope of this issue.